internal/util/tar.go

```diff
@@ -0,0 +1,66 @@
+package util
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+)
+
+// FSToTarGZ writes the filesystem represented by fsys to w as a gzipped tar archive.
+// This function unsets user and group information in the tar archive so that readers
+// of archives produced by this function do not need to account for differences in
```

Comment:
This should help alleviate some of the issues we see when attempting to prune red hat catalog images that include root-permission directories. Maybe this is worth highlighting in the user-facing documentation? If it's not overly technical

Reply:
Yep, +1 on that recommendation.

Reply:
It might be a little bit technical given the existing way most users would interact with this (`rukpakctl run`). That subcommand does all the necessary tar.gz stuff under the hood and it is not exposed at all to users.

Regardless, I could definitely see value in documenting the upload manager service as a separate thing because there may be in-cluster processes that can directly interact with the upload manager without requiring the extra port-forward stuff that `rukpakctl run` provides.